mlir/include/Quantum/IR/QuantumOps.td

```diff
@@ -341,7 +338,8 @@ class UnitaryGate_Op<string mnemonic, list<Trait> traits = []> :
     let extraClassDeclaration = extraBaseClassDeclaration;
 }
 
-def SetStateOp : Gate_Op<"set_state", []> {
+def SetStateOp : Gate_Op<"set_state",
```

Comment:
All gates should be pure (due the qubit value semantics).

Although, do the memrefs need a read?